src/theory/shared_solver.cpp

```diff
@@ -104,9 +105,13 @@ EqualityStatus SharedSolver::getEqualityStatus(TNode a, TNode b)
   return EQUALITY_UNKNOWN;
 }
 
-void SharedSolver::sendLemma(TrustNode trn, TheoryId atomsTo)
+bool SharedSolver::propagateLit(TNode predicate, bool value)
 {
-  d_te.lemma(trn, LemmaProperty::NONE, atomsTo);
+  if (value)
+  {
+    return d_out.propagate(predicate);
+  }
+  return d_out.propagate(predicate.notNode());
```

Comment:
```suggestion
  return d_out.propagate(value ? predicate : predicate.notNode());
```

Reply:
This is actually not possible due to Node vs TNode issues; notNode creates a Node, thus this saves ref counting as I understand it.  This pattern is used throughout the code in other places for propagations, e.g. https://github.com/cvc5/cvc5/blob/master/src/theory/theory_eq_notify.h#L41.